src/AllocArrays.jl

```diff
@@ -27,13 +27,15 @@ const CURRENT_ALLOCATOR = ScopedValue{Allocator}(DEFAULT_ALLOCATOR)
 # PrecompileTools workload
 @setup_workload begin
     @compile_workload begin
-        b = BumperAllocator(2^10) # 1 KiB
-        a = AllocArray([1])
-        c = CheckedAllocArray(a, MemValid(true))
-        with_allocator(b) do
-            similar(a) .= 1
-            similar(c) .= 1
-            reset!(b)
+        for alloc in (() -> BumperAllocator(), () -> BumperAllocator(2^10)) # (SlabBuffer, 1 KiB)
+            b = alloc()
+            a = AllocArray([1])
```

Comment:
unrelated to this PR but I guess
```suggestion
            a = AllocArray([1.0f0])
```
might help compile out some float32 stuff for downstream use